impl/client/src/main/javascript/web/dojo/pentaho/common/PageControl.js

```diff
@@ -30,161 +30,185 @@
+                    a11yUtil.makeAccessibleActionButton(this.nextPageControl));
+            },
+
+            registerLocalizationLookup: function (f) {
+                this.getLocaleString = f;
+                this._localize();
+            },
+
+            registerPageNumberChangeCallback: function (f) {
+                this.changePageNumberCallback = f;
+            },
+
+            setPageNumber: function (pageNumber) {
+                if (pageNumber < 1 || pageNumber > this.pageTotal) {
+                    // TODO Alert user of invalid entry
+                    this._resetPageNumber();
+                    return;
+                }
+                try {
+                    this._updateInternalPageNumber(pageNumber);
+                } catch (err) {
+                    console.error("Unable to update page number: " + err);
+                    this._resetPageNumber();
+                }
+            },
+
+            getPageNumber: function () {
+                return this.pageNumberInput.get("value");
+            },
+
+            getPageTotal: function () {
+                return this.pageTotal;
+            },
+
+            setPageCount: function (pageCount) {
+                if (typeof (pageCount) == 'number') {
+                    this.pageTotal = pageCount;
+                } else {
+                    this.pageTotal = 1; // Reset to default
+                }
+                this.pageTotalSpan.innerHTML = '/ ' + pageCount;
+                this._updatePageButtonState();
+            },
+
+            reset: function () {
+                this._updateInternalPageNumber(1);
+            },
+
+            _localize: function () {
+                this.prevPageControl.setAttribute("title", this.getLocaleString("PageControlPreviousPage_title"));
+                this.nextPageControl.setAttribute("title", this.getLocaleString("PageControlNextPage_title"));
+            },
+
+            _updatePageButtonState: function () {
+                this.pagePrevDisabled = this.pageNumber === 1;
+                this.pageNextDisabled = this.pageTotal === this.pageNumber;
+                if (this.pagePrevDisabled) {
+                    domClass.add(this.prevPageControl, "pc_pagePrevDisabled");
+                    this.prevPageControl.setAttribute("tabindex", "-1");
+                } else {
+                    domClass.remove(this.prevPageControl, "pc_pagePrevDisabled");
+                    this.prevPageControl.setAttribute("tabindex", "0");
+                }
+                if (this.pageNextDisabled) {
+                    domClass.add(this.nextPageControl, "pc_pageNextDisabled");
+                    this.nextPageControl.setAttribute("tabindex", "-1");
+                } else {
+                    domClass.remove(this.nextPageControl, "pc_pageNextDisabled");
+                    this.nextPageControl.setAttribute("tabindex", "0");
+                }
```

Comment:
Regarding this, that understand to be the implementation of the roving tab index navigation pattern, in the context of being a child of toolbar, I believe that this would be better implemented by the toolbar controller itself, as noted elsewhere, where a similar implementation was proposed: https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1677#discussion_r1333285474.